torchvision/transforms/transforms.py

```diff
@@ -570,7 +576,7 @@ def get_params(img, scale, ratio):
             w = int(round(math.sqrt(target_area * aspect_ratio)))
             h = int(round(math.sqrt(target_area / aspect_ratio)))
 
-            if random.random() < 0.5:
+            if random.random() < 0.5 and min(ratio) <= (w / h) <= max(ratio):
```

Comment:
Since we're going to swap w and h, NEW aspect ratio (h / w) should be checked.
```suggestion
            if random.random() < 0.5 and min(ratio) <= (h / w) <= max(ratio):
```